common/presenters/move-to-message-banner-component.js

```diff
@@ -0,0 +1,147 @@
+const { isEmpty, kebabCase } = require('lodash')
+
+const i18n = require('../../config/i18n')
+const filters = require('../../config/nunjucks/filters')
+const componentService = require('../services/component')
+
+const frameworkToTaskListComponent = require('./framework-to-task-list-component')
+
+function _returnComponent(title, content) {
+  return {
+    allowDismiss: false,
+    classes: 'app-message--instruction govuk-!-padding-right-0',
+    title: {
+      text: title,
+    },
+    content: {
+      html: content,
+    },
+  }
+}
+
+function moveToMessageBannerComponent({
+  move,
+  moveUrl,
+  canAccess,
+  assessmentType,
+}) {
+  const { profile, status } = move
+  const { id: profileId, [assessmentType]: assessment } = profile || {}
+  const baseUrl = `${moveUrl}/${kebabCase(assessmentType)}`
+  const context = assessmentType
+
+  if (['proposed'].includes(status) || !profileId) {
+    return undefined
+  }
+
+  let title
+  let content
+
+  // No PER exists
```

Comment:
I wonder if this could be broken up into smaller presenters that are then assembled in this parent one. There's already a fairly clear separation which you've had to describe with comments, this could be even clearer if it was code.

e.g. 
```
if (perStatusUtils.noPer(status, assessment)) {
  return renderNoPer(any, params, that, are, needed
}

if ((perStatusUtils.perConfirmed(status, assessment)) {
  return renderConfirmedPer(other, params, that, are, needed)
}
```

This might also deal with the CodeClimate complaints about complexity and return statements.

Reply:
What Robin said.

Also, I'm a bit concerned about all the generation of html within the presenter. Personally I would prefer to have that lifting done by an actual template. (and not have logic in our templates ;) )

Reply:
I've made an attempt at this approach with the latest commit. Let me know your thoughts.